> It looks like mypy on Debian, Ubuntu, and Fedora doesn't actually use a discrete "typeshed" package even if available, and rather bundles typeshed as `mypy.typeshed`. Would it be possible to use that instead, since your long-term goal is to use mypy anyway?

From what I can tell `mypy.typeshed` only has `stdlib` types and not typeshed's stubs folders. From testing this without installing `python3-typeshed` or `types-pyyaml` mypy raises the following.

```
src/rclpy/rclpy/rclpy/parameter.py:27: error: Library stubs not installed for "yaml"  [import-untyped]
src/rclpy/rclpy/rclpy/parameter.py:27: note: Hint: "python3 -m pip install types-PyYAML"
```

--- a/rosdep/python.yaml
+++ b/rosdep/python.yaml
@@ -9519,9 +9519,16 @@ python3-typing-extensions:
     bionic:
       pip:
         packages: [typing-extensions]
-python3-typeshed:
+python3-types-pyyaml:
   debian: [python3-typeshed]
-  ubuntu: [python3-typeshed]
+  ubuntu: 
+    '*': [python3-typeshed]
+    focal:
+      pip:
+        packages: [types-pyyaml]
+  rhel:
+    pip:
+      packages: [types-pyyaml]
 python3-tz:
   alpine: [py3-tz]
   arch: [python-pytz]

--- a/rosdep/python.yaml
+++ b/rosdep/python.yaml
@@ -9505,6 +9505,19 @@ python3-typeguard-pip:
   ubuntu:
     pip:
       packages: [typeguard]
+python3-types-pyyaml:
+  debian: [python3-typeshed]
+  fedora:
+    pip:
+      packages: [types-pyyaml]
+  ubuntu: 
+    '*': [python3-typeshed]
+    focal:
+      pip:
+        packages: [types-pyyaml]
+  rhel:
+    pip:
+      packages: [types-pyyaml]
 python3-typing-extensions:
   arch: [python-typing_extensions]
   debian: [python3-typing-extensions]
@@ -9519,16 +9532,6 @@ python3-typing-extensions:
     bionic:
       pip:
         packages: [typing-extensions]
-python3-types-pyyaml:
-  debian: [python3-typeshed]
-  ubuntu: 
-    '*': [python3-typeshed]
-    focal:
-      pip:
-        packages: [types-pyyaml]
-  rhel:
-    pip:
-      packages: [types-pyyaml]
 python3-tz:
   alpine: [py3-tz]
   arch: [python-pytz]

--- a/rosdep/python.yaml
+++ b/rosdep/python.yaml
@@ -9507,9 +9507,7 @@ python3-typeguard-pip:
       packages: [typeguard]
 python3-types-pyyaml:
   debian: [python3-typeshed]
-  fedora:
-    pip:
-      packages: [types-pyyaml]
+  fedora: [python3-types-pyyaml]
   ubuntu: 
     '*': [python3-typeshed]
     focal:

